runtime/src/chplsys.c

```diff
@@ -85,12 +85,12 @@ size_t chpl_getHeapPageSize(void) {
     else {
 
       size_t tmpPageSize;
-      int  num_scanned;
-      char units;
+      int scanCnt;
+      char units[4]; // leave room for terminating null byte
 
-      if ((num_scanned = sscanf(ev, "%zi%c", &tmpPageSize, &units)) != 1) {
-        if (num_scanned == 2 && strchr("kKmMgG", units) != NULL) {
-          switch (units) {
+      if ((scanCnt = sscanf(ev, "%zd%1[kKmMgG]", &tmpPageSize, units)) > 0) {
```

Comment:
@panzone - can you revert this change to chplsys.c - it's in "fix stack overflow and...". I'll make this change in a separate PR - your PR need not be concerned with it.